exosys/src/main.rs

```diff
@@ -89,38 +117,192 @@ async fn dump_finalized_headers(rpc_client: &Client<Config>) -> () {
     }
 }
 
-#[tokio::main(worker_threads = 8)]
-async fn main() -> Result<()> {
+/// Supply address with port if needed.
+///
+/// Transform address as it is displayed to user in <https://polkadot.js.org/>
+/// to address with port added if necessary that could be fed to `jsonrpsee`
+/// client.
+///
+/// The port is set here to default 443 if there is no port specified in
+/// address itself, since default port in `jsonrpsee` is unavailable for now.
+///
+/// See for details <https://github.com/paritytech/jsonrpsee/issues/554`>
+///
+/// Some addresses have port specified, and should be left as is.
+fn address_with_port(str_address: &str) -> String {
+    match PORT.captures(str_address) {
+        Some(caps) => {
+            if caps.name("port").is_some() {
+                str_address.to_string()
+            } else {
+                match caps.name("tail") {
+                    Some(tail) => format!("{}:443{}", &caps["body"], tail.as_str()),
+                    None => format!("{}:443", &caps["body"]),
+                }
+            }
+        }
+        None => str_address.to_string(),
+    }
+}
+
+#[tokio::main]
```

Comment:
Quote from https://docs.rs/tokio/latest/tokio/attr.main.html:

> Note: This macro can be used on any function and not just the main function. Using it on a non-main function makes the function behave as if it was synchronous by starting a new runtime each time it is called. If the function is called often, it is preferable to create the runtime using the runtime builder so the runtime can be reused across calls.

Would not it be simpler to put it once on `async fn main` and `.await` calling these functions from there?

Reply:
Sure. I've just quickly slapped together chunks of other projects to get the result; I mean to run 2 or 3 threads there concurrently later. It's ugly everywhere at the moment.